astropy/io/ascii/html.py

```diff
@@ -83,7 +90,10 @@ def process_lines(self, lines):
                                         'installed to read HTML tables')
 
         if 'parser' not in self.html:
-            soup = BeautifulSoup('\n'.join(lines))
+            if lxml_installed:
+                soup = BeautifulSoup('\n'.join(lines),"lxml")
+            else:
+                soup = BeautifulSoup('\n'.join(lines),"html.parser")
```

Comment:
Incidentally this patch is *almost* fine modulo some stylistic bits:
1) For consistency's sake, rename `lxml_installed` to `HAVE_LXML`
2) Add spaces after commas between function arguments.

And then rebase or whatever to remove the unrelated change in `astropy.units`.  Also re-title the PR to what it's actually addressing.